sample-app/app/jobs/test_job.rb

```diff
@@ -4,5 +4,8 @@ class TestJob < ApplicationJob
 
   def perform(*args)
     puts "Job performed with args: #{args}"
+    if args[0].is_a?(Hash) && args[0][:error]
```

Comment:
Probably not necessary, or if you want to keep, maybe a TestErrorJob or also place this code in async.